### ISSUE TITLE
Comments are not properly managed by ServiceUtil

### ISSUE THREAD

**Issue by esteban-aliverti** [quarkusio/quarkus#2892]
**Describe the bug**
`ServiceUtil` is used by `UndertowBuildStep` to extract the list of class named from all the `META-INF/services/javax.servlet.ServletContainerInitializer` files present in the classpath. 
The problem is that if one of these `javax.servlet.ServletContainerInitializer` files contains lines with comments (`#`), the lines are still parsed and their content tried to be used as a valid class name down the road.

The end result is an exception similar to this:

```
15:28:32,043 ERROR [io.qua.dev.DevModeMain] Failed to start quarkus: java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.BeanArchiveProcessor#build threw an exception: java.lang.IllegalStateException: Failed to index: #
	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:129)
	at io.quarkus.dev.DevModeMain.doStart(DevModeMain.java:157)
	at io.quarkus.dev.DevModeMain.main(DevModeMain.java:84)
Caused by: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
	[error]: Build step io.quarkus.arc.deployment.BeanArchiveProcessor#build threw an exception: java.lang.IllegalStateException: Failed to index: #
	at io.quarkus.builder.Execution.run(Execution.java:108)
	at io.quarkus.builder.BuildExecutionBuilder.execute(BuildExecutionBuilder.java:121)
	at io.quarkus.deployment.QuarkusAugmentor.run(QuarkusAugmentor.java:97)
	at io.quarkus.runner.RuntimeRunner.run(RuntimeRunner.java:94)
	... 2 more
Caused by: java.lang.IllegalStateException: Failed to index: #
	at io.quarkus.deployment.index.IndexingUtil.indexClass(IndexingUtil.java:35)
	at io.quarkus.arc.deployment.BeanArchiveProcessor.build(BeanArchiveProcessor.java:63)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at io.quarkus.deployment.ExtensionLoader$1.execute(ExtensionLoader.java:507)
	at io.quarkus.builder.BuildContext.run(BuildContext.java:398)
	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2011)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1538)
	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1429)
	at java.lang.Thread.run(Thread.java:748)
	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: java.io.IOException: Stream closed
	at java.io.BufferedInputStream.getInIfOpen(BufferedInputStream.java:159)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
	at java.io.DataInputStream.readFully(DataInputStream.java:195)
	at java.io.DataInputStream.readFully(DataInputStream.java:169)
	at org.jboss.jandex.Indexer.verifyMagic(Indexer.java:1288)
	at org.jboss.jandex.Indexer.index(Indexer.java:1587)
	at io.quarkus.deployment.index.IndexingUtil.indexClass(IndexingUtil.java:32)
	... 13 more
```

**Expected behavior**
- Lines that start with a `#` should be omitted.
- Lines containing a class name followed by a comment should be parsed in a way so only the part before the comment is used.

**Actual behavior**
Comments are treated as a class name leading to the exception described above.

**To Reproduce**
1. Create a sample quarkus app:
```
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create -DprojectGroupId=org.acme  -DprojectArtifactId=getting-started -DclassName="org.acme.quickstart.GreetingResource" -Dpath="/hello"
```
2. Include the following dependency in the application's pom.xml
```
<dependency>
    <groupId>org.apache.logging.log4j</groupId>
    <artifactId>log4j-web</artifactId>
    <version>2.7</version>
</dependency>
```
This dependency includes a `javax.servlet.ServletContainerInitializer` file containing some comments.
3. Try to run the application: `mvn compile quarkus:dev`

**Environment:**
 - Output of `uname -a` or `ver`: `Linux esteban-xps 4.4.0-148-generic #174-Ubuntu SMP Tue May 7 12:20:14 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`
 - Output of `java -version`: 
```
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (build 1.8.0_212-8u212-b03-0ubuntu1.16.04.1-b03)
OpenJDK 64-Bit Server VM (build 25.212-b03, mixed mode)
```
 - Quarkus version or git rev: `999-SNAPSHOT`

**Comment by gsmet** [comment]
@esteban-aliverti would you be interested in fixing it?

**Comment by esteban-aliverti** [comment]
Sure. I already have one ;).
I'm preparing the PR right now.

**Comment by gsmet** [comment]
Cool, thanks.

Tests would be welcome too as we all thought it was already properly handled.

**Comment by esteban-aliverti** [comment]
A PR can be found here: https://github.com/quarkusio/quarkus/pull/2893

**Comment by esteban-aliverti** [comment]
Thank you for the support. I think this issue can be closed now.

**Comment by machi1990** [comment]
Fixed by #2893. 

Thanks @esteban-aliverti
